lib/connections/limiter.go

```diff
@@ -99,50 +188,66 @@ func (lim *limiter) String() string {
 
 // limitedReader is a rate limited io.Reader
 type limitedReader struct {
-	reader  io.Reader
-	limiter *limiter
-	isLAN   bool
+	reader   io.Reader
+	limiter  *limiter
+	isLAN    bool
+	remoteID protocol.DeviceID
 }
 
 func (r *limitedReader) Read(buf []byte) (int, error) {
 	n, err := r.reader.Read(buf)
 	if !r.isLAN || r.limiter.limitsLAN.get() {
-		take(r.limiter.read, n)
+		deviceLimiter, ok := r.limiter.deviceReadLimiter[r.remoteID]
+		if ok == false {
+			l.Debugln("deviceLimiter was not in the map")
+			deviceLimiter = rate.NewLimiter(rate.Inf, limiterBurstSize)
```

Comment:
How can this happen?

Reply:
It happens after removing remote device. I'm not sure about limitedWriter but I added it just to be safe.